website/docs/r/volume.html.markdown

```diff
@@ -0,0 +1,46 @@
+---
+layout: "cloudscale"
+page_title: "cloudscale.ch: cloudscale_volume"
+sidebar_current: "docs-cloudscale-resource-volume"
+description: |-
+  Provides a cloudscale.ch Volume (block storage) resource. This can be used to create, modify, and delete Volumes.
+---
+
+# cloudscale\_volume
+
+Provides a cloudscale.ch Volume (block storage) resource. This can be used to create, modify, and delete Volumes.
+
+## Example Usage
+
+```hcl
+# Create a new Server
+resource "cloudscale_server" "web-worker01" {
+  name        = "web-worker01"
+  flavor_slug = "flex-4"
+  image_slug  = "debian-9"
+  ssh_keys    = ["ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIL2jzgla23DfRVLQr3KT20QQYovqCCN3clHrjm2ZuQFW user@example.com"]
+}
+
+# Add a Volume to web-worker01
+resource "cloudscale_volume" "web-worker01-vip" {
+  name         = "web-worker-data"
+  size_gb      = 100
+  type         = "ssd"
+  server_uuids = ["${cloudscale_server.web-worker01.id}"]
+}
+```
+
+## Argument Reference
+
+The following arguments are supported when creating/changing Volumes:
+
+* `name` - (Required) Name of the new volume.
```

Comment:
Is there a restriction on the name? Can it be an arbitrary length? Is the name limited to a certain set of characters?

Reply:
It can be arbitrary. The length is limited, but not to something "small", so I would vote for leaving it like that. The API documentation is a bit more precise for quite a few things (we've chosen to do it like that when contributing `cloudscale_server`). It's valid criticism, though.